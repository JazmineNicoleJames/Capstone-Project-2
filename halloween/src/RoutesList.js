import React, {useContext} from "react";
import { Route, Routes } from "react-router-dom";
import YouTubeVideo from "./YouTubeVideo";
import AllCostumes from "./AllCostumes";
import Questions from "./Questions";
import CostumeDetail from "./CostumeDetail";


/* RoutesList component
    Routes to AllCostumes component "/costumes",
        * to CostumeDetail component "/costumes/:item_name",
        * to YouTubeVideo component "/youtube-video/:item_name",
        * and to Questions component "/"
*/

const RoutesList = ({allCostumes}) => {

    return (
        <Routes>
            <Route path="/costumes" element={<AllCostumes allCostumes={allCostumes} />} />
            <Route path="/costumes/:item_name" element={<CostumeDetail allCostumes={allCostumes}/>} />
            <Route path="/youtube-video/:item_name" element={<YouTubeVideo />} />
            <Route path="/" element={<Questions />} />
        </Routes>
    )
}
export default RoutesList;