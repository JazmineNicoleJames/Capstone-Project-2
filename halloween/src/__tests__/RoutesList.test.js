import React from "react";
import { render, waitFor } from '@testing-library/react';
import { MemoryRouter } from "react-router-dom";
import RoutesList from '../RoutesList';

test("should match snapshot", async () => {
    const { asFragment } = render(
        <MemoryRouter>
            <RoutesList />
        </MemoryRouter>
    );
    await waitFor(() => {
      expect(asFragment()).toMatchSnapshot();
    });
});


test("should render costume '/costume' route path", () => {
    const mockCostumes = [
        {item_name: "fake"}
    ];

    const { getByText } = render(
        <MemoryRouter initialEntries={['/costumes']}>
            <RoutesList allCostumes={mockCostumes}/>
        </MemoryRouter>
    );
    const costumeElement = getByText("fake");
    
    expect(costumeElement).toBeInTheDocument();
});


test("should render home '/' route path", () => { 
    const { getByTestId } = render(
        <MemoryRouter initialEntries={['/']}>
            <RoutesList/>
        </MemoryRouter>
    );
    const form = getByTestId('my-form');

    expect(form).toBeInTheDocument();
})
