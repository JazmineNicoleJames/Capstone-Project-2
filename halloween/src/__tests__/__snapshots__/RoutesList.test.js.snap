// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1
      class="welcome-msg"
    >
      Welcome. Let's come up with a costume, shall we?
    </h1>
    <h3
      class="directions"
    >
       Begin by answering the following prompts 
    </h3>
    <form
      class="custom-select"
      data-testid="my-form"
    >
      <div
        class="custom-select"
      >
        <label
          for="theme"
        >
          <select
            id="theme"
          >
            <option
              value="theme"
            >
              Select a theme
            </option>
            <option
              value="classic"
            >
              Classic
            </option>
            <option
              value="pop-culture"
            >
              Pop Culture
            </option>
            <option
              value="animal"
            >
              Animal
            </option>
          </select>
        </label>
      </div>
      <div
        class="custom-select"
      >
        <label
          for="genderPref"
        >
          <select>
            <option
              value="genderPref"
            >
              Select a gender preference
            </option>
            <option
              value="feminine"
            >
              Feminine
            </option>
            <option
              value="masculine"
            >
              Masculine
            </option>
            <option
              value="gender-neutral"
            >
              Gender Neutral
            </option>
          </select>
        </label>
      </div>
      <div
        class="custom-select"
      >
        <label
          for="experienceLevel"
        >
          <select>
            <option
              value="experienceLevel"
            >
               Select your experience Level
            </option>
            <option
              value="easy"
            >
              Easy
            </option>
            <option
              value="intermediate"
            >
              Intermediate
            </option>
            <option
              value="advanced"
            >
              Advanced
            </option>
          </select>
        </label>
      </div>
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
